### PR TITLE
258 replace bible api with usfm

### PR DESCRIPTION
**Pull request by lyonsil**
This removes all usage of bible-api.com from our project.  Where we used it to load scripture we now use the USFM data provider.  Where we used it as a "random" target for `fetch` calls we now use `example.com`.

It's worthwhile to note that the ParatextData USFM loader doesn't use full English book names but instead has 3 letter names/IDs for each book.  No effort was made to translate full English names to the 3 letter ones for now.  I am assuming that other controls will do that mapping and pass down what ParatextData is expecting.  If we want the USFM data provider to work from more names, we can follow up with another commit to add more functionality at the data provider level inside of C#.

I tacked on the change to `README.md` as part of this PR.

### CODE REVIEW

**Review by tjcouch-sil — APPROVED** [review]
Awesome! Thanks for doing this. Would you mind also cleaning up `paranext-extension-template` as well? Part of me wonders if we're getting to the point where we just need to rip everything out of there and go ahead and make an examples repo... Or something along those lines. But that's a worry for another issue ;)

> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276#-:-NZSp_9_3oF6ANelK-sm:b-t67xjm)* status: all files reviewed, 1 unresolved discussion (waiting on @lyonsil)

___
*[`extensions/lib/quick-verse/quick-verse.ts` line 171 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZSoFTE3llpPxaM1_rm:-NZSoFTE3llpPxaM1_rn:b9u54a2) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/extensions/lib/quick-verse/quick-verse.ts#L171)):*
> ```XML
>     // Just get notifications of updates with the 'notify' selector
>     if (verseRef === 'notify') return undefined;
>     const selector = this.#getSelector(verseRef.trim());
> ```

Can we trim inside `#getSelector` since it is the standardizing method? Would be good to make sure everyone is doing the same thing with the selectors.

**Review by FoolRunning — CHANGES_REQUESTED** [review]
> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276)* status: all files reviewed, 3 unresolved discussions (waiting on @lyonsil and @tjcouch-sil)

___
*[`extensions/lib/quick-verse/quick-verse.ts` line 171 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZSoFTE3llpPxaM1_rm:-NZaC0lp9ayV1K8mnWo4:b-hagxgo) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/extensions/lib/quick-verse/quick-verse.ts#L171)):*
<details><summary><i>Previously, tjcouch-sil (TJ Couch) wrote…</i></summary><blockquote>

Can we trim inside `#getSelector` since it is the standardizing method? Would be good to make sure everyone is doing the same thing with the selectors.
</blockquote></details>

Why are we trimming at all? It seems like needing a trim here means something is supplying an unexpected verse reference format. Wherever that verse reference is coming from seems like it should be fixed there. 
___
*[`src/renderer/testing/test-buttons-panel.component.tsx` line 136 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZaCpRs-Mfx4LwfHL5T:-NZaCpRs-Mfx4LwfHL5U:btl0k3v) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/src/renderer/testing/test-buttons-panel.component.tsx#L136)):*
> ```XML
>   );
> 
>   const [verseRef, setVerseRef] = useState<string>('Jhn 11:35');
> ```

Ideally, the book in the verse reference should be all-caps: ``JHN 11:35``. That is the correct way for the 3-letter codes to be written (I'm actually surprised it works without them being caps).
___
*[`src/renderer/testing/test-quick-verse-heresy-panel.component.tsx` line 12 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZaD8kNB1KsdNGER4th:-NZaD8kNB1KsdNGER4ti:b2nreh6) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/src/renderer/testing/test-quick-verse-heresy-panel.component.tsx#L12)):*
> ```XML
> 
> export function TestQuickVerseHeresyPanel() {
>   const [verseRef, setVerseRef] = useState<string>('Jhn 11:35');
> ```

Same as above.

**Review by lyonsil — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276)* status: 7 of 9 files reviewed, 2 unresolved discussions (waiting on @FoolRunning and @tjcouch-sil)

___
*[`extensions/lib/quick-verse/quick-verse.ts` line 171 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZSoFTE3llpPxaM1_rm:-NZaOVQh-l5hIKwiaBgG:benb1q9) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/extensions/lib/quick-verse/quick-verse.ts#L171)):*
<details><summary><i>Previously, FoolRunning (Tim Steenwyk) wrote…</i></summary><blockquote>

Why are we trimming at all? It seems like needing a trim here means something is supplying an unexpected verse reference format. Wherever that verse reference is coming from seems like it should be fixed there. 
</blockquote></details>

Done - Note that the verseRef might actually be "latest" or something like that at first.

For now the selectors come from a text box, and while testing I noticed somehow I got an extra space at the beginning.  It was confusing at first.  I can drop it if desired, but I think it only helps users.
___
*[`src/renderer/testing/test-buttons-panel.component.tsx` line 136 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZaCpRs-Mfx4LwfHL5T:-NZaP3ne3KbAzmLOEPEJ:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/src/renderer/testing/test-buttons-panel.component.tsx#L136)):*
<details><summary><i>Previously, FoolRunning (Tim Steenwyk) wrote…</i></summary><blockquote>

Ideally, the book in the verse reference should be all-caps: ``JHN 11:35``. That is the correct way for the 3-letter codes to be written (I'm actually surprised it works without them being caps).
</blockquote></details>

Done.
___
*[`src/renderer/testing/test-quick-verse-heresy-panel.component.tsx` line 12 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZaD8kNB1KsdNGER4th:-NZaP4ipDJ1e_YvjKElg:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/src/renderer/testing/test-quick-verse-heresy-panel.component.tsx#L12)):*
<details><summary><i>Previously, FoolRunning (Tim Steenwyk) wrote…</i></summary><blockquote>

Same as above.
</blockquote></details>

Done.

**Review by lyonsil — COMMENTED** [review]
Sure, I can do that in a separate PR.  I haven't cloned that repo yet, so I'll need to set that up.

> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276#-:-NZaPYmtD2qA8mAIZYIl:bgyrcoj)* status: 7 of 9 files reviewed, 2 unresolved discussions (waiting on @FoolRunning and @tjcouch-sil)

**Review by tjcouch-sil — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276#-:-NZaWyyS0kPViZtIy40v:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @FoolRunning)

**Review by FoolRunning — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276)* status: all files reviewed, 1 unresolved discussion (waiting on @lyonsil)

___
*[`extensions/lib/quick-verse/quick-verse.ts` line 171 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZSoFTE3llpPxaM1_rm:-NZaXtbE9v2EKsaPES1r:birb66p) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/extensions/lib/quick-verse/quick-verse.ts#L171)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

Done - Note that the verseRef might actually be "latest" or something like that at first.

For now the selectors come from a text box, and while testing I noticed somehow I got an extra space at the beginning.  It was confusing at first.  I can drop it if desired, but I think it only helps users.
</blockquote></details>

In case I wasn't clear, what I meant was that if the textbox is creating invalid verse references, then where that text is read, the trim should be done there and not at this low level (i.e. it should be done at the UI layer in this case so all the lower-level code doesn't constantly have to trim everywhere where the verse reference is used).
___
*[`src/renderer/testing/test-buttons-panel.component.tsx` line 136 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZaCpRs-Mfx4LwfHL5T:-NZaYLHiEw0nefF7JrLy:b1ungmo) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/src/renderer/testing/test-buttons-panel.component.tsx#L136)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

According to the Reviewable diff tool, this doesn't actually look done in this location. Am I missing something?

**Review by lyonsil — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276)* status: all files reviewed, 1 unresolved discussion (waiting on @FoolRunning)

___
*[`extensions/lib/quick-verse/quick-verse.ts` line 171 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZSoFTE3llpPxaM1_rm:-NZaZWwO71h-otPbh_u2:b-rjick6) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/extensions/lib/quick-verse/quick-verse.ts#L171)):*
<details><summary><i>Previously, FoolRunning (Tim Steenwyk) wrote…</i></summary><blockquote>

In case I wasn't clear, what I meant was that if the textbox is creating invalid verse references, then where that text is read, the trim should be done there and not at this low level (i.e. it should be done at the UI layer in this case so all the lower-level code doesn't constantly have to trim everywhere where the verse reference is used).
</blockquote></details>

We are using a vanilla text box at the moment.  When we switch to a "proper" selector I agree it should format things properly.  I also doubt that `quick-verse` will be a real data provider in Paratext.Bible (or at least not close to its current implementation), as we don't want to be reporting and tracking heresy.  I also doubt that we want a vanilla text box UI selector.

Basically this is all just sample/text code from my POV.  If we want to production-ize it, that's probably a separate ticket.
___
*[`src/renderer/testing/test-buttons-panel.component.tsx` line 136 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZaCpRs-Mfx4LwfHL5T:-NZa_-vQ8uQcBWysSZZV:bz86v9k) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/src/renderer/testing/test-buttons-panel.component.tsx#L136)):*
<details><summary><i>Previously, FoolRunning (Tim Steenwyk) wrote…</i></summary><blockquote>

According to the Reviewable diff tool, this doesn't actually look done in this location. Am I missing something?
</blockquote></details>

My bad.  I had updated the file in VSCode, but I didn't hit save.  Done for real now.

**Review by tjcouch-sil — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/276)* status: 8 of 9 files reviewed, 1 unresolved discussion (waiting on @FoolRunning)

___
*[`extensions/lib/quick-verse/quick-verse.ts` line 171 at r1](https://reviewable.io/reviews/paranext/paranext-core/276#-NZSoFTE3llpPxaM1_rm:-NZaYqNQBsPDY-Uvn7Wj:bhwu73e) ([raw file](https://github.com/paranext/paranext-core/blob/cefa755b2efc2c40a1ae86487bb2ccc2adf24671/extensions/lib/quick-verse/quick-verse.ts#L171)):*
<details><summary><i>Previously, lyonsil (Matt Lyons) wrote…</i></summary><blockquote>

We are using a vanilla text box at the moment.  When we switch to a "proper" selector I agree it should format things properly.  I also doubt that `quick-verse` will be a real data provider in Paratext.Bible (or at least not close to its current implementation), as we don't want to be reporting and tracking heresy.  I also doubt that we want a vanilla text box UI selector.

Basically this is all just sample/text code from my POV.  If we want to production-ize it, that's probably a separate ticket.
</blockquote></details>

I'd think in general simple reformatting might best belong on the highest layer of the backend as possible. It's likely many different people will be ingesting this data provider, and it would probably be a good user experience if we do easy reformats right as things come in.

It's also worth noting that in my opinion true validation to make sure that input is not going to do something evil (and again in my opinion we might as well lump in general input validation where it's simple enough like in this case) should be done at least on the backend; preferably on the frontend too but at least backend. Since we have a bunch of unknown code hitting our apis, we are likely best to distrust stuff hitting our apis. But I guess that's a security best practice, so maybe we will or won't do that depending on what we decide to do.
